apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/queries.ex

```diff
@@ -806,42 +736,35 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Queries do
     value
   end
 
-  def retrieve_endpoint_values(client, device_id, interface_descriptor, mapping) do
-    query_statement =
-      prepare_get_property_statement(
-        mapping.value_type,
-        false,
-        interface_descriptor.storage,
-        interface_descriptor.storage_type
-      )
+  def retrieve_property_values(realm, device_id, interface_descriptor, mapping) do
+    %InterfaceDescriptor{
+      storage_type: :multi_interface_individual_properties_dbtable,
+      interface_id: interface_id
+    } = interface_descriptor
 
-    query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(query_statement)
-      |> DatabaseQuery.put(:device_id, device_id)
-      |> DatabaseQuery.put(:interface_id, interface_descriptor.interface_id)
-      |> DatabaseQuery.put(:endpoint_id, mapping.endpoint_id)
+    %Mapping{endpoint_id: endpoint_id, value_type: value_type} = mapping
+    column_name = astarte_type_to_db_column_name(value_type)
 
-    DatabaseQuery.call!(client, query)
-  end
+    keyspace_name =
+      CQLUtils.realm_name_to_keyspace_name(realm, Config.astarte_instance_id!())
 
-  defp prepare_get_property_statement(
-         value_type,
-         metadata,
-         table_name,
-         :multi_interface_individual_properties_dbtable
-       ) do
-    metadata_column =
-      if metadata do
-        ",metadata"
-      else
-        ""
-      end
+    all_values =
+      IndividualProperty
+      |> where(device_id: ^device_id, interface_id: ^interface_id, endpoint_id: ^endpoint_id)
+      |> put_query_prefix(keyspace_name)
```

Comment:
The previous query was targeting the `interface_descriptor.storage` table instead of the `individual_properties` table directly: is that correct?

Besides, I believe this chunk of code is missing a call to `Repo` APIs to actually fetch data.

Reply:
Good catch!

The query targets `IndividualProperty` because it's only used in the case of fetching (device-owned) (individual) properties. However we might find a way to keep it generic as before